File: java/src/main/java/io/fivexer/sdk/model/WorkerDevice.java
package io.fivexer.sdk.model;

/**
 * A registered Expo push token for the native app.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerDevice {
    private String token;
    private String platform;
    private String createdAt;

    public String getToken() { return token; }
    public String getPlatform() { return platform; }
    public String getCreatedAt() { return createdAt; }
}

File: php/src/Model/WorkerDevice.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A registered Expo push token for the native app.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerDevice
{
    public function __construct(
        public readonly string $token,
        /** 'ios' | 'android' | 'unknown' */
        public readonly string $platform,
        public readonly ?string $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            token: (string) ($data['token'] ?? ''),
            platform: (string) ($data['platform'] ?? ''),
            createdAt: isset($data['createdAt']) ? (string) $data['createdAt'] : null,
        );
    }
}
